src/circular_buffer.jl

```diff
@@ -1,44 +1,50 @@
-
 """
 New items are pushed to the back of the list, overwriting values in a circular fashion.
 """
-type CircularBuffer{T} <: AbstractVector{T}
+type CircularBuffer{T, N} <: AbstractArray{T, N}
     capacity::Int
     first::Int
-    buffer::Vector{T}
+    buffer::Array{T, N}
+    used::Int
+    colons::Vector{Colon}
+
+    function CircularBuffer(capacity::Int, others_dims...)
+        a = Array{T, N}(capacity, others_dims...)
```

Comment:
For context, I didn't do this originally because the capacity might have been much larger than the typical size. Allocating all of the capacity is only better when you actually fill the buffer.